File: Scala/1047.scala
import scala.io.StdIn.readLine

object Main {
  def main(args: Array[String]) {
    val Array(h1, m1, h2, m2) = readLine().split(" ").map(_.toInt)

    var hh = h2
    var mm = m2
    if (h2 <= h1 && m2 <= m1) {
      hh += 24
    } else if (m2 <= m1) {
      mm += 60
      hh -= 1
    }

    val t = (hh * 3600) + (mm * 60) - (h1 * 3600) - (m1 * 60)
    val h = t / 3600
    val m = (t % 3600) / 60

    printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", h, m)
  }
}

File: 1. Beginner/1047/1047.dart
import 'dart:io';

void main() {
    var input = stdin.readLineSync();
    var values = input.split(' ').map(int.parse).toList();
    
    var h1 = values[0];
    var m1 = values[1];
    var h2 = values[2];
    var m2 = values[3];

    if (h2 <= h1 && m2 <= m1) {
        h2 += 24;
    } else if (m2 <= m1) {
        m2 += 60;
        h2 -= 1;
    }

    var t = ((h2 * 3600) + (m2 * 60)) - ((h1 * 3600) + (m1 * 60));
    var h = t ~/ 3600;
    var m = (t - (h * 3600)) ~/ 60;

    print("O JOGO DUROU $h HORA(S) E $m MINUTO(S)");
}